frontend/src/metabase/parameters/utils/mapping-options.ts

```diff
@@ -61,37 +68,28 @@ function buildTextTagOption(tagName) {
   };
 }
 
-/**
- *
- * @param {import("metabase-lib/metadata/Metadata").default} metadata
- * @param {import("metabase-types/api").Parameter|null} parameter
- * @param {import("metabase-types/api").Card} card
- * @param {import("metabase-types/store").StoreDashcard|null} [dashcard]
- * @returns {*}
- */
 export function getParameterMappingOptions(
-  question,
-  parameter = null,
-  card,
-  dashcard = null,
+  question: Question,
+  parameter: Parameter | null | undefined = null,
+  card: Card,
+  dashcard: BaseDashboardCard | null | undefined = null,
 ) {
-  if (dashcard && ["heading", "text"].includes(card.display)) {
+  if (dashcard && isVirtualDashCard(dashcard)) {
     const tagNames = tag_names(dashcard.visualization_settings.text || "");
     return tagNames ? tagNames.map(buildTextTagOption) : [];
   }
```

Comment:
IIRC tag names are only available for `heading` and `text` dashcards

Reply:
right, I'll add this check back.

`type VirtualCardDisplay = "text" | "action" | "link" | "heading"`

Reply:
[da7ab9e](https://github.com/metabase/metabase/pull/38596/commits/da7ab9ea3336a83768116aa339d26f1f132acfa5)